### ISSUE TITLE
Questions about precision and implementation

### ISSUE THREAD

**Issue by KengChiLiu** [marvis/pytorch-yolo2#18]
There are two questions~

First, I'm training your code  with batch size=16, learning rate=0.00025 and do follow the instructions (train on VOC2007+2012 and evaluate on VOC2007 by your code). However, the precision does not meet the result you show. Mine is only about 0.47.

epoch ,       precision,                      recall,                    fscore 
50,         0.424161073509,    0.472817954719,   0.447164825738
100,       0.473204017118,    0.51305070614,     0.492317426357
150,       0.477074824322,    0.51463009101,     0.495136367244

Final training error : 
x(0.244189232588)
y(0.235951408744)
w(0.0568502359092)
h(0.577977240086)
conf(0.695957958698)
cls(4.76443195343)
total(6.57535791397)

Second  is that I have fed the weights provided by the author of yolo which is trained on COCO and VOC into your code. However, I got a totally different result between yours and author's (with same weight file), different objects detected and different confidence. By the way, I have changed your threshold to the same as author's, but still different. Do you have any idea what's going on ?

For instance, after i fed the same image A, I got

YOLOv2 : person: 26%, person: 69%, person: 70%, person: 65%, person: 70%  
YOURS : bus: 0.599364, person: 0.598274, person: 0.542617, person: 0.329442, person: 0.315203, person: 0.245802

Both above were test on "same weight file" trained on COCO.

**Comment by marvis** [comment]
1. When you decrease batch size, you may need to decrease lr and increase decay. You can have a try.
https://github.com/marvis/pytorch-yolo2/blob/master/train.py#L105

2.  Do you provide the correct cfg?

**Comment by KengChiLiu** [comment]
1. I did decrease the learning rate. By the way, I found that the precision show by test() in "train.py" is 0.477 but MAP calculated by "scripts/voc_eval.py" is 0.72.  Is this normal???  Shouldn't they be the same??? 
Additionally, your calculation on MAP of released model is different from the paper, but as far as I know, the definition of MAP is the same. How come they are different?

2. I did provide the correct cfg. (Yours cfg are the same as author's)
Still different result......

**Comment by marvis** [comment]
1. the accuracy in train.py is different from that in voc_eval.py. I never reach the accuracy as the paper describes.
2. does the voc model correct?

**Comment by KengChiLiu** [comment]
1. So precision=0.4-0.5 seems to be normal ??? Do you remember the precision that you get???
2.  Yes. All settings are the same.

**Comment by marvis** [comment]
1. a little larger than 0.5 
2. the results couldn't be exactly the same. Because reorg layer is implemented differently.

**Comment by KengChiLiu** [comment]
My precision was 0.56 at the first epoch and 0.47 at final. However, it is obvious that the model is still poor at the beginning of training. So, I'm wondering the meaning of the precision. Have you met this before?

**Comment by marvis** [comment]
the precison is for detection accuracy without consideration of classification

**Comment by KengChiLiu** [comment]
However, getting a high precision at the beginning seems to be abnormal...

**Comment by wwxJream** [comment]
@KengChiLiu Hey bro! Can you get 72 mAP ?? How to do that? I only get 59 mAP when use `yolo-voc.cfg` to train. Could you help me? Thanks!!
